### PR TITLE
Make `x2/y2/latitude2/longitude2` no longer have a `type` and `bin`

### PR DESCRIPTION
**Pull request by kanitw**
(as the `type` should be exactly the same as `x/y/latitude/longitude`)

- Remove `normalizeRangedUnit`, which is no longer necessary
- Simplify code / test for `errorbar.ts`
- Change `tooltipForChannelDefs` to `tooltipForEncoding`

### CODE REVIEW

**Comment by kanitw** [comment]
I'm also adding @chanwutk to review as removing type actually simply many logic for error bars.

File: build/vega-lite-schema.json [hunk 1]
```diff
@@ -3888,6 +3888,55 @@
...
       "type": "object"
     },
+    "FieldDefWithoutType": {
+      "additionalProperties": false,
+      "properties": {
+        "aggregate": {
+          "$ref": "#/definitions/Aggregate",
+          "description": "Aggregation function for the field\n(e.g., `mean`, `sum`, `median`, `min`, `max`, `count`).\n\n__Default value:__ `undefined` (None)"
+        },
+        "bin": {
```
Comment (domoritz) [inline build/vega-lite-schema.json]: Could `x2` ever be binned differently from `x`?

File: build/vega-lite-schema.json [hunk 2]
```diff
@@ -3888,6 +3888,55 @@
...
+            {
+              "type": "string"
+            },
+            {
+              "$ref": "#/definitions/RepeatRef"
+            }
+          ],
+          "description": "__Required.__ A string defining the name of the field from which to pull a data value\nor an object defining iterated values from the [`repeat`](https://vega.github.io/vega-lite/docs/repeat.html) operator.\n\n__Note:__ Dots (`.`) and brackets (`[` and `]`) can be used to access nested objects (e.g., `\"field\": \"foo.bar\"` and `\"field\": \"foo['bar']\"`).\nIf field names contain dots or brackets but are not nested, you can use `\\\\` to escape dots and brackets (e.g., `\"a\\\\.b\"` and `\"a\\\\[0\\\\]\"`).\nSee more details about escaping in the [field documentation](https://vega.github.io/vega-lite/docs/field.html).\n\n__Note:__ `field` is not required if `aggregate` is `count`."
+        },
+        "timeUnit": {
```
Comment (domoritz) [inline build/vega-lite-schema.json]: Same question as for bin.

File: build/vega-lite-schema.json (hunk 1 above)
Comment (kanitw) [reply build/vega-lite-schema.json]: Good point. Actually, we shouldn't even support `bin` for `x2`.  (If `x` is binned, `x2` is inherently the bin end.)

File: build/vega-lite-schema.json (hunk 2 above)
Comment (kanitw) [reply build/vega-lite-schema.json]: In practice, I don't see why it should be different.  But It's probably feasible to create a weird plot that have them differ.

File: build/vega-lite-schema.json (hunk 1 above)
Comment (domoritz) [reply build/vega-lite-schema.json]: Then the name of the field def needs to be something else but I think this will make things much cleaner. So 👍

**Comment by kanitw** [comment]
> Just a question about binning and time units but otherwise LGTM.

At this point, given that `aggregate` is the only property that can differ, I wonder if the alternative approach to have `field2` and `aggregate2` for `x` / `y` as suggested in #4423 is more sensible? 

That said, we still have to support the old syntax otherwise it will be quite a massive breaking change.

File: build/vega-lite-schema.json (hunk 2 above)
Comment (domoritz) [reply build/vega-lite-schema.json]: Hmm, timeUnit is usually like bin so I'd think it's the same as bin. Should we just disallow this until someone complains?

**Comment by domoritz** [comment]
What about a chart where you make the start of a bar a fixed value and the end data dependent. That wouldn't be possible with `aggregate2`.

**Comment by kanitw** [comment]
> What about a chart where you make the start of a bar a fixed value and the end data dependent. That wouldn't be possible with aggregate2.

Good point.  We may consider `aggregate2` and `field2` as shorthands in the future then.  But we definitely need to keep `x2` and `y2` as the primary way to specify this.

File: examples/compiled/area_temperature_range.vg.json [hunk 3]
```diff
@@ -12,7 +12,7 @@
       "url": "data/seattle-weather.csv",
       "format": {
         "type": "csv",
-        "parse": {"date": "date", "temp_max": "number", "temp_min": "number"}
+        "parse": {"date": "date", "temp_max": "number"}
```
Comment (kanitw) [inline examples/compiled/area_temperature_range.vg.json]: Need to still include `temp_min`

File: examples/compiled/interactive_brush.vg.json [hunk 4]
```diff
@@ -295,7 +295,7 @@
             {"value": "grey"}
           ],
           "tooltip": {
-            "signal": "{\"Horsepower\": format(datum[\"Horsepower\"], \"\"), \"Miles_per_Gallon\": format(datum[\"Miles_per_Gallon\"], \"\"), \"Cylinders\": ''+datum[\"Cylinders\"]}"
+            "signal": "{\"Horsepower\": format(datum[\"Horsepower\"], \"\"), \"Miles_per_Gallon\": format(datum[\"Miles_per_Gallon\"], \"\")}"
```
Comment (kanitw) [inline examples/compiled/interactive_brush.vg.json]: Somehow Cylinders is now missing

**Comment by domoritz** [comment]
> Good point. We may consider aggregate2 and field2 as shorthands in the future then. But we definitely need to keep x2 and y2 as the primary way to specify this.

Maybe later. I'm not yet convinced we want it until we have thoroughly considered the design.

File: build/vega-lite-schema.json (hunk 2 above)
Comment (domoritz) [reply build/vega-lite-schema.json]: Agreed

File: src/encoding.ts [hunk 5]
```diff
@@ -67,7 +73,8 @@ export interface Encoding<F> {
   /**
    * Secondary error value of x coordinates for error specified `"errorbar"` and `"errorband"`.
    */
-  xError2?: FieldDefWithoutScale<F> | ValueDef;
+  // `xError2` cannot have type as it should have the same type as `xError`
```
Comment (domoritz) [inline src/encoding.ts]: `xError` should have the same type as `x`. Therefore `xError` should also be a `SecondaryRangeFieldDef`

Comment (domoritz) [reply src/encoding.ts]: Please fix in a follow up PR.

**Pull request merged**